### PR TITLE
ADR 107: Rename proto versions to pre-v1 betas

### PR DESCRIPTION
**Pull request by mzabaluev**
Improvement proposal for #95 to make protobuf versioning cleaner on the way to the 1.0 release.

[Rendered](https://github.com/cometbft/cometbft/blob/mikhail/adr-betaize-proto-versions/docs/architecture/adr-107-betaize-proto-versions.md)

### CODE REVIEW

File: docs/architecture/adr-107-betaize-proto-versions.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+(started with [#736][cometbft#736]) have been confined to latter versions
+of the proto packages in order to preserve source code compatibility
+and ease migration for developers. The earlier packages, therefore, do not
+constitute exemplary protobuf material and may even be rejected by a schema
+repository linter.
+
+## Alternative Approaches
+
+We can do nothing and go ahead with the current versioning rework as per
+[ADR-103], which does solve the main problem of managing backward-incompatible
```
Comment (adizere) [inline docs/architecture/adr-107-betaize-proto-versions.md:32]: Nit: If this is intended to hyperlink to the reference introduced on L81, it doesn't work. Same for the link on L53.

**Review by adizere — COMMENTED** [review]
Makes sense!

File: docs/architecture/adr-107-betaize-proto-versions.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# ADR 107: Rename proto versions preceding 1.0 to pre-v1 betas
+
+## Changelog
+
+- 2023-07-11: Initial draft
```
Comment (thanethomson) [inline docs/architecture/adr-107-betaize-proto-versions.md:5]:
```suggestion
- 2023-07-11: Initial draft (@mzabaluev)
```

File: docs/architecture/adr-107-betaize-proto-versions.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+## Status
+
+Proposed (CHANGE BEFORE MERGING)
```
Comment (thanethomson) [inline docs/architecture/adr-107-betaize-proto-versions.md:9]:
```suggestion
Accepted
```

File: docs/architecture/README.md [hunk 4]
```diff
@@ -43,6 +43,7 @@ numbering our ADRs from 100 onwards.
 ### Proposed
 
 - [ADR-103: Protobuf definition versioning](./adr-103-proto-versioning.md)
+- [ADR-107: Rename protobuf versions of 0.x releases to pre-v1 betas](./adr-107-betaize-proto-versions.md)
```
Comment (thanethomson) [inline docs/architecture/README.md:46]: This should be under the "Accepted" section, as with ADR-103 (I'll move ADR-103 in a separate PR).

**Review by thanethomson — APPROVED** [review]

**Comment by sergio-mena** [comment]
I know I'm getting late to the party.
In principle, the modifications proposed in this ADR make sense to me. 
However, it is not clear to me what will happen, in detail, at the moment we release `v1`. What will happen with the structures that don't change at that point? Will they still refer to `beta` versions? Will we "copy-paste" them into `v1` versions, so that our version history doesn't refer back to `beta` versions?

**Pull request merged**